### ISSUE TITLE
No active class in Permissions

### ISSUE THREAD

**Issue by Quy** [joomla/joomla-cms#15214]
### Steps to reproduce the issue
Edit a category.
Go to Permissions tab.
If you do see an active active, close it and try a different record. 
Other Permissions sections are affected too.

![permissions-no-active](https://cloud.githubusercontent.com/assets/368084/24887842/f3bcaddc-1e13-11e7-83eb-6b1dec13707e.jpg)
In Firebug, you can see no `active` class.
![permissions-firebug](https://cloud.githubusercontent.com/assets/368084/24887865/0e1d49a2-1e14-11e7-8135-2718f7cf2237.jpg)

### Additional comments
Tested in Firefox 53.0, Chrome 58.0.30, and Edge 38.143 in Windows 10 Home.

**Comment by franz-wohlkoenig** [comment]
Can't confirm Issue:
![bildschirmfoto 2017-04-11 um 07 32 00](https://cloud.githubusercontent.com/assets/8235763/24894218/28e49176-1e89-11e7-8917-c7eefcbbcadc.JPG)

**Comment by C-Lodder** [comment]
@Quy - is your issue regarding Joomla 3.6.5 or 3.7.0-rc1?

**Comment by Quy** [comment]
Latest staging version as 4/11.

**Comment by Quy** [comment]
This issue is related to PR #13572. Reverted `tabs-state.js` from v3.6.5 and it fixed this issue.

**Comment by C-Lodder** [comment]
If you have any issues for J4, please be sure to put [4.0] before the title so they can easily be found.

**Comment by Quy** [comment]
It is for v3.7.

@Bakual Please confirm issue. Thanks.

**Comment by Bakual** [comment]
I guess my change only revealed another issue. Since previously, the tabstate once set would be present in eternity, now you get the effect each time after you have closed the browser. The issue was probably there before, but only once and you don't remember it anymore.

Imho, the permissions should activate a tab by default which then is changed later by the script (if a state is stored). Instead, no tab is active by default which is the real issue here.

**Comment by Quy** [comment]
The following code sets the initial active default tab, but tabstate clears it I assume.

```
        foreach ($groups as $group)
        {
            // Initial Active Tab
            $active = "";

            if ($group->value == 1)
            {
                $active = "active";
            }

            $html[] = '<li class="' . $active . '">';
            $html[] = '<a href="#permission-' . $group->value . '" data-toggle="tab">';
            $html[] = str_repeat('<span class="level">&ndash;</span> ', $curLevel = $group->level) . $group->text;
            $html[] = '</a>';
            $html[] = '</li>';
        }
```

**Comment by Bakual** [comment]
Hmm, it may be related to https://github.com/joomla/joomla-cms/pull/12503 then, as that one actually changed the JS. My PR only changed the storage place an shouldn't have such side effects.

After playing around a bit, it looks like the page stores two different tab states, once for the main tabs and another one for the permission tab. They may overwrite eachother if I see that right since they both use the same key.
@dgt41 That PR was from you, can you have a look?

**Comment by dgt41** [comment]
@Bakual is the id the same for all instances of permissions?
If so we can add some more logic to the script to manage this case.
I will look at this tonight

**Comment by Bakual** [comment]
@dgt41 Looks like it uses the same key as the tabs from the edit form itself, but the value is eg `["#permission-6"]` instead of the `["#page-permissions"]` from the edit form. They indeed overwrite eachother now. So i think we need a way for the permission tabs to have a different key than the edit form itself. Same issue would happen to any form where there are multiple tabs.

**Comment by PhilETaylor** [comment]
Fixed with https://github.com/joomla/joomla-cms/pull/15748 please test

**Comment by franz-wohlkoenig** [comment]
closed as having PR #15748<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15214">issues.joomla.org/tracker/joomla-cms/15214</a>.</sub>
